fix: hide insights submenu for free users

KartikLabhshetwar opened the pull request:
## What does this PR do?

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

- Fixes #25711
- Fixes [CAL-6885](https://linear.app/calcom/issue/CAL-6885/multiple-tabs-directing-to-a-single-page)

## Visual Demo (For contributors especially)

#### Video Demo (if applicable):

before:

https://github.com/user-attachments/assets/628c35e4-c630-4178-93a7-9f4781cfe8b8

free plan:


https://github.com/user-attachments/assets/c4145c4c-2af3-4973-b6e2-ebb55f8e7c37



after:


https://github.com/user-attachments/assets/39270490-5abe-41a2-9e9d-e4839ad1c39e




## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. Write details that help to start the tests -->

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Checklist

<!-- Remove bullet points below that don't apply to you -->

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings










































<!-- This is an auto-generated description by cubic. -->
---
## Summary by cubic
Hide the Insights submenu for free plan users and route the Insights nav to the Insights page. This aligns access with plan and avoids multiple tabs pointing to a single page (Linear CAL-6885).

- **Bug Fixes**
  - Gated Insights with useHasPaidPlan and hide the submenu while plan status is pending to prevent flash.
  - Removed Insights child items for free plans and set href to /insights.
  - Kept full submenu and direct routes for paid plans.

<sup>Written for commit 5d89304987e482a85816e6099c95fc58bdf1e50b. Summary will update automatically on new commits.</sup>

<!-- End of auto-generated description by cubic. -->

KartikLabhshetwar committed:
fix: hide insights submenu for free users

KartikLabhshetwar committed:
Merge branch 'main' into fix/hide-insights-submenu-free-plan

KartikLabhshetwar committed:
fix: allow navigation access while plan pending

KartikLabhshetwar committed:
Merge branch 'main' into fix/hide-insights-submenu-free-plan

KartikLabhshetwar committed:
Merge branch 'main' into fix/hide-insights-submenu-free-plan

dhairyashiil reviewed (changes requested):
On the Insights page, I notice the flow starts directly with creating a team. Let’s show this page first, and only start the team creation flow when the user clicks “Create Team.”

<img width="2560" height="1440" alt="Screenshot 2025-12-10 at 4 41 40 PM (2)" src="https://github.com/user-attachments/assets/d032038b-ca7a-4e0a-aaec-72cc74e420dc" />

KartikLabhshetwar committed:
Merge branch 'main' into fix/hide-insights-submenu-free-plan

KartikLabhshetwar committed:
refactor: update upgrade link to insights in navigation

KartikLabhshetwar committed:
Merge branch 'main' into fix/hide-insights-submenu-free-plan

KartikLabhshetwar commented:
> On the Insights page, I notice the flow starts directly with creating a team. Let’s show this page first, and only start the team creation flow when the user clicks “Create Team.”
> 
> <img alt="Screenshot 2025-12-10 at 4 41 40 PM (2)" width="2000" height="1440" src="https://private-user-images.githubusercontent.com/93669429/524782289-d032038b-ca7a-4e0a-aaec-72cc74e420dc.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjUzNjYyODEsIm5iZiI6MTc2NTM2NTk4MSwicGF0aCI6Ii85MzY2OTQyOS81MjQ3ODIyODktZDAzMjAzOGItY2E3YS00ZTBhLWFhZWMtNzJjYzc0ZTQyMGRjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEyMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMjEwVDExMjYyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAyNTIwYWRiMDA1M2QzOWQxNmJmMTg1MmI1N2Y5MmM2MWEzM2JiM2M2NDI3NzgwMGIzZGQyNjdhZGNiMDk0ZWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.7kGYL-SJ9b3HMPjRsAuJgGki-uapAITVVuvJJHUUBh0">

hi, @dhairyashiil done with the changes as required.


https://github.com/user-attachments/assets/423bd019-7f83-4737-9fb8-785cd1c71871

KartikLabhshetwar committed:
Merge branch 'main' into fix/hide-insights-submenu-free-plan

KartikLabhshetwar committed:
Merge branch 'main' into fix/hide-insights-submenu-free-plan

KartikLabhshetwar committed:
Merge branch 'main' into fix/hide-insights-submenu-free-plan

KartikLabhshetwar committed:
Merge branch 'main' into fix/hide-insights-submenu-free-plan

KartikLabhshetwar committed:
Merge branch 'main' into fix/hide-insights-submenu-free-plan

KartikLabhshetwar committed:
Merge branch 'main' into fix/hide-insights-submenu-free-plan

KartikLabhshetwar committed:
Merge branch 'main' into fix/hide-insights-submenu-free-plan

KartikLabhshetwar committed:
Merge branch 'main' into fix/hide-insights-submenu-free-plan

KartikLabhshetwar commented:
hi @dhairyashiil @anikdhabal @volnei can anyone please review this pr?

KartikLabhshetwar committed:
Merge branch 'main' into fix/hide-insights-submenu-free-plan

dhairyashiil commented on packages/features/shell/navigation/Navigation.tsx:
```diff
@@ -16,7 +17,24 @@ import { NavigationItem, MobileNavigationItem, MobileNavigationMoreItem } from "
+  nextPathname: string;
+}) => Boolean(prevPathname?.startsWith("/bookings/")) && nextPathname.startsWith("/bookings/");
+
+type NavigationInsightsConfig = {
```
this type is no longer necessary

Consider removing this type and just passing a simple hasInsightsAccess: boolean to getNavigationItems().

dhairyashiil reviewed (changes requested):
Left some comments, please address

KartikLabhshetwar committed:
Merge branch 'main' into fix/hide-insights-submenu-free-plan

KartikLabhshetwar committed:
fix: addressed all the issues

KartikLabhshetwar committed:
Merge branch 'main' into fix/hide-insights-submenu-free-plan

KartikLabhshetwar commented:
> Left some comments, please address

hi @dhairyashiil, thank you so much for the review, sorry for making such simple mistakes, i have addressed all of the issues mentioned here.

KartikLabhshetwar committed:
Merge branch 'main' into fix/hide-insights-submenu-free-plan

dhairyashiil committed:
fix: prevent flash of premium features for free users during loading

Change from optimistic to pessimistic loading so free users don't
briefly see the Insights submenu while plan status is being fetched.

dhairyashiil reviewed (approved):
Thanks for the PR! After testing again locally I noticed that Free users briefly see the full Insights submenu for a moment before it disappears. I made a small change to fix that.

I’ve attached before/after videos so you can see the behavior I was referring to.
Thank you again for the contribution.

Before:


https://github.com/user-attachments/assets/4c466482-017d-4c34-ad3f-1490efb9ea17




After:



https://github.com/user-attachments/assets/1f2b3dea-059a-4808-86f7-da281a3e461d

KartikLabhshetwar committed:
Merge branch 'main' into fix/hide-insights-submenu-free-plan
